### DISCUSSION TITLE
logout other logged in user after change his password by admin

### DISCUSSION THREAD

**Discussion by mshannaq** [codeigniter4/shield#984] [Q&A]
In my project I add the ability to the supseradmin user to change any user password

e.g:

```php
$user = $this->usermodel->findById(123);
$filled_data = [
'password' => 'newpassword',
'username' => 'newusername',
'email' => 'newemail@example.com'
];
$user->fill($filled_data);
// forget any remember me tokens
auth()->forget($user);
```

that works fine and password changed . but the effected user will still logged in.

I want to make sure that if user  (that I changed his password) currently logged in then perform a logout for him.

is there any way (ready to use in shiled) to do that?

**Comment by kenjis** [comment]
> is there any way (ready to use in shiled) to do that?

No. 
Why do you need it?

**Comment by mshannaq** [comment]
> > is there any way (ready to use in shiled) to do that?
> 
> No. Why do you need it?

while admin manage the users he may be need to change some user password.

and if he changed password for user and the user is logged in he will still logged in

**Reply by datamweb** [reply]
In general, it is not for the admin to change the password of other users. I strongly disagree with this type of implementation.
You must provide the possibility to change the password only by the user himself.

**Reply by kenjis** [reply]
Yes, if an admin changes a user's password for some reasons, the user should update their password immediately.
See https://shield.codeigniter.com/user_management/forcing_password_reset/

**Comment by mshannaq** [comment]
@datamweb sometimes may be the user cannot access his account and cannot receive emails to reset the password, so there is an option to let the admin change his account password (and set to force password reset after login [good idea])

I took a look to shield and see if admin ban user it will be logged out immediately even if the user logged in

so can I do that if password changed also?

**Reply by kenjis** [reply]
It is a trick. It does not log the user out immediately, but it logs the user out right after the next access.
You can implement the same thing when password changed.

See https://github.com/codeigniter4/shield/blob/a0ec47fd7dfc2116cbdf62ebca4a0592906d10b0/src/Filters/SessionAuth.php#L62-L68

**Reply by mshannaq** [reply]
this will logout the admin who changed the password for a user. not the user which I have changes his password.

**Reply by kenjis** [reply]
Sorry, I don't get what you say. I showed the code to ban a user as a reference.
The code does not log out admin, but log the banned user out.
You must implement what you want by yourself.
